plugins/woocommerce/src/Internal/ComingSoon/ComingSoonAdminBarBadge.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\ComingSoon;
+
+/**
+ * Adds hooks to add a badge to the WordPress admin bar showing site visibility.
+ */
+class ComingSoonAdminBarBadge {
+
+	/**
+	 * Sets up the hooks.
+	 *
+	 * @internal
+	 */
+	final public function init() {
+		add_action( 'admin_bar_menu', array( $this, 'site_visibility_badge' ), 31 );
+		add_action( 'wp_head', array( $this, 'output_css' ) );
+		add_action( 'admin_head', array( $this, 'output_css' ) );
+	}
+
+	/**
+	 * Invalidate the WordPress object cache and other known caches.
+	 *
+	 * @internal
+	 * @param WP_Admin_Bar $wp_admin_bar The WP_Admin_Bar instance.
+	 */
+	public function site_visibility_badge( $wp_admin_bar ) {
+		$labels = array(
+			'coming-soon'       => __( 'Coming soon', 'woocommerce' ),
+			'store-coming-soon' => __( 'Store coming soon', 'woocommerce' ),
+			'live'              => __( 'Live', 'woocommerce' ),
+		);
+
+		if ( get_option( 'woocommerce_coming_soon' ) === 'yes' ) {
+			if ( get_option( 'woocommerce_store_pages_only' ) === 'yes' ) {
+				$key = 'store-coming-soon';
+			} else {
+				$key = 'coming-soon';
+			}
+		} else {
+			$key = 'live';
+		}
+
+		$args = array(
+			'id'    => 'woocommerce-site-status-badge',
+			'title' => $labels[ $key ],
+			'href'  => admin_url( 'admin.php?page=wc-settings&tab=site-visibility' ),
+			'meta'  => array(
+				'class' => 'woocommerce-site-status-badge-' . $key,
+			),
+		);
+		$wp_admin_bar->add_node( $args );
+	}
+
+	/**
+	 * Invalidate the WordPress object cache and other known caches.
+	 *
+	 * @internal
+	 */
+	public function output_css() {
+		if ( is_admin_bar_showing() ) {
+			echo '<style>
```

Comment:
I was on the fence whether to embed this or add it to a new file. Either way, we don't want it to be included when the admin bar isn't showing for customers / anon users.